### PR TITLE
Remove project content from test images

### PR DESCRIPTION
**Pull request by 8-bit-fox**
#### Description
This PR aims at removing the project code from the test images. Reasons have been explained in #2344 .

### CODE REVIEW

**Comment by tbabej** [comment]
Welcoming this initiative, it could make our tests run faster too. While the images are currently written in such a way that they would reuse cached layers locally (and hence would not need to re-install OS packages each time), that does not seem to be happening in Travis by default.

Another way to achieve this would be to push the built test images and then use `--cache-from something/test-taskwarrior:ubuntu1804` to reuse the OS installation layers.

**Comment by sruffell** [comment]
@8-bit-fox I think a change like this should go into the 2.6.0 branch first and then be backported to 2.5.2 if necessary. 

Otherwise, I have https://github.com/sruffell/taskwarrior/commits/wip-remove-project-content as an example, that does the centos7 and ubuntu1804 images, but otherwise I think it should wait for #2349 to be merged first before this is merged in.

**Comment by 8-bit-fox** [comment]
In general, I'd agree. However, in #2339 @pbeckingham suggested that hewould cherry-pick commits from 2.6.0 to 2.5.2 and rather create a 3.0.0. Thus I am finebasing it on master, but 2.6.0 just seems rebase overhead to me.

But despite that - is there any reason to have two different PRs? Rebasing and changing the draft PR is a matter of seconds.

**Comment by sruffell** [comment]
> In general, I'd agree. However, in #2339 @pbeckingham suggested that hewould cherry-pick commits from 2.6.0 to 2.5.2 and rather create a 3.0.0. Thus I am finebasing it on master, but 2.6.0 just seems rebase overhead to me.

Maybe I'm confused, but it looks like your PR is scheduled to go into 2.5.2?  The current head of development is 2.6.0 is shouldn't new work go there and then be backported to the stable branch as necessary (in this case 2.5.2)

Do you have a link to the specific comment where @pbeckingham said he would cherry-pick from 2.6.0 onto 2.5.2 (which, again, seems to suggest that changes should go into 2.6.0)?  I couldn't find it when I did a cursory scan of #2339.

**Comment by 8-bit-fox** [comment]
> Do you have a link to the specific comment where @pbeckingham said he would cherry-pick from 2.6.0 onto 2.5.2 (which, again, seems to suggest that changes should go into 2.6.0)? I couldn't find it when I did a cursory scan of #2339.

Ah, my bad. I mistyped the issue number. It was #2338 . On mobile you don't get the pop-up with the issue titles and I didn't confirm. The specific comment is [here](https://github.com/GothenburgBitFactory/taskwarrior/issues/2338#issuecomment-729067633).

**Comment by 8-bit-fox** [comment]
> Maybe I'm confused, but it looks like your PR is scheduled to go into 2.5.2? The current head of development is 2.6.0. Shouldn't new work go into 2.6.0 and be backported to the stable branch as necessary (in this case 2.5.2)?

I must admit, I do not understand the branching strategy here yet. The common ancestor of master, 2.5.2 and 2.6.0 is  [Post251ToMaster](8f09a35ad20e94b7418ce2de4e875db261a6d397). According to the comment, it rather looks as if 2.5.2 would be the future and 2.6.0 would deprecate in favor of 3.0.0

IMO the only question is if SemVer is implemented here. If so, a major or minor version change would be the place to put this refactoring. Or - it is regarded as a bug fix (coupling of test infrastructure and project code being the bug :wink: ) so we should put it to 2.5.2 :slightly_smiling_face:

**Comment by tbabej** [comment]
@8-bit-fox TW tries to follow semantic versioning. That said, general CI improvement like this one will be merged into all active development branches. As @sruffell mentions, the 2.6.0 is the development head, so there is a slight preference for this MR to be pointed towards 2.6.0, and it will be backported to 2.5.2 at the time of merge.

> Thus I am finebasing it on master, but 2.6.0 just seems rebase overhead to me.

As README mentions (let us know if it is not clear!), `master` in this project is actually last stable release. In this particular case that's end of 2.5.1 branch, so basing this MR on master would actually create more work (technically it would be quite difficult because Travis CI was never even introduced on 2.5.1 :slightly_smiling_face: ).

With respect to 2.6.0 / 3.0.0 discussion, there is a chance that 2.6.0 will be renamed to 3.0.0 at some point in the future, but that is I think irrelevant to the discussion at hand. The branch contant will stay the same during the change.

TL;DR: As a general rule, 2.6.0 should work as the target branch. But really, for the purpose of this MR, it does not matter much.

**Comment by 8-bit-fox** [comment]
Thanks, @tbabej , for clarification!

**Comment by 8-bit-fox** [comment]
@tbabej any reason for closing?

**Comment by tbabej** [comment]
I don't remember doing so, I think this got automatically closed by Github because we deleted the target branch 2.5.2 after its release.

**Comment by tbabej** [comment]
@8-bit-fox I am not able to change the target branch it seems, maybe you can? We should definitely reopen this PR or open its equivalent against 2.6.0. Good catch.

**Comment by 8-bit-fox** [comment]
I'll reopen it once I had the chance to rebase.
